kazoo/protocol/connection.py

```diff
@@ -478,10 +478,30 @@ def zk_loop(self):
             self.client._session_callback(KeeperState.CLOSED)
             self.logger.log(BLATHER, 'Connection stopped')
 
+    def _expand_client_hosts(self):
+        # Expand the entire list in advance so we can randomize it if needed
+        host_ports = []
+        for host, port in self.client.hosts:
+            try:
+                for rhost in socket.getaddrinfo(host.strip(), port, 0, 0, socket.IPPROTO_TCP):
+                    host_ports.append((rhost[4][0], rhost[4][1]))
+            except socket.gaierror:
+                # Skip hosts that don't resolve
```

Comment:
What do u think about logging this? Seems like a warning would at least be useful (as someone giving an unresolveable host seems not that good).